tests/test_checkout_integration.py

```diff
@@ -110,3 +112,115 @@ def test_checkout_flow(product_in_stock, client):
+    value = signing.get_cookie_signer(salt=Cart.COOKIE_NAME).sign(cart.token)
+    client.cookies[Cart.COOKIE_NAME] = value
+
+    # Go to cart page
+    cart_page = client.get(urls['cart'])
+    cart_lines = cart_page.context['cart_lines']
+    assert len(cart_lines) == cart.lines.count()
+    assert cart_lines[0]['variant'] == variant
+    # Enter checkout
+    checkout_index = client.get(urls['checkout_index'])
+    # Checkout index redirects directly to shipping address step
+    assert_redirect(checkout_index, urls['checkout_shipping_address'])
+    shipping_address = client.get(urls['checkout_shipping_address'])
+    assert shipping_address.status_code == 200
+    # Enter shipping address data
+    shipping_data = {'address': user_shipping_address.pk}
+    shipping_response = client.post(
+        urls['checkout_shipping_address'], data=shipping_data)
+    # Select shipping method
+    assert_redirect(shipping_response, urls['checkout_shipping_method'])
+    shipping_method_page = client.get(urls['checkout_shipping_method'])
+    assert shipping_method_page.status_code == 200
+    # Redirect to summary after shipping method selection
+    shipping_method_response = client.post(
+        urls['checkout_shipping_method'], data={'method': shipping_method.pk})
+    assert_redirect(shipping_method_response, urls['checkout_summary'])
+    # Summary page asks for Billing address, default is the same as shipping
+    summary_response = client.post(urls['checkout_summary'],
+                                   data={'address': 'shipping_address'})
+    # After summary step, order is created and it waits for payment
+    order = Order.objects.latest('pk')
+    order_payment_url = reverse('order:payment', kwargs={'token': order.token})
+    assert_redirect(summary_response, order_payment_url)
+    payment_method_page = client.get(order_payment_url)
+    assert payment_method_page.status_code == 200
+    # Select payment method
+    payment_page = client.post(order_payment_url, data={'method': 'default'},
+                               follow=True)
+    assert len(payment_page.redirect_chain) == 1
+    assert payment_page.status_code == 200
+    # Go to payment details page, enter payment data
+    payment_page_url = payment_page.redirect_chain[0][0]
+    payment_data = {
+        'status': 'preauth',
+        'fraud_status': 'unknown',
+        'gateway_response': '3ds-disabled',
+        'verification_result': 'waiting'}
+    payment_response = client.post(payment_page_url, data=payment_data)
+    assert payment_response.status_code == 302
+    # Target page contains full URL with domain from Site object
+    site = Site.objects.get_current()
+    order_details = reverse('order:details', kwargs={'token': order.token})
+    expected_url = 'http://%s%s' % (site, order_details)
+    assert payment_response['Location'] == expected_url
+    order_details_page = client.get(order_details)
```

Comment:
I'd stop the test case here. Nothing below is in any way related to the checkout process.